src/components/qas/questions/qaCalculators/kineticEnergyCalculator/index.jsx

```diff
@@ -0,0 +1,207 @@
+  }
+
+  checkAnswerable(answerable, active) {
+    if (!answerable && active) {
+      this.setState(prevState => ({
+        ...prevState,
+        answerable: true,
+      }));
+    }
+  }
+
+  calculateKineticEnergy({ mass, velocity }) {
+    if (!mass || !velocity) return null;
+
+    return 0.5 * mass * velocity ** 2;
+  }
+
+  getNewVal(value, valType) {
+    const { value: oldValue } = this.state;
+
+    if (!value || !valType) return oldValue;
+
+    const newVal = {
+      ...oldValue,
+      [valType]: value,
+    };
+
+    return {
+      ...newVal,
+      kineticEnergy: this.calculateKineticEnergy(newVal),
+    };
+  }
+
+  handleFocus = () => {
+    this.setState(prevState => ({
+      ...prevState,
+      hasFocus: true,
+    }));
+  };
+
+  handleBlur = () => {
+    this.setState(prevState => ({
+      ...prevState,
+      hasFocus: false,
+    }));
+  };
+
+  handleChange = (value, valType) => {
+    const { question, answerHandler } = this.props;
+    const { id } = question;
+
+    this.setState(
+      prevState => ({
+        ...prevState,
+        hasFocus: true,
+        value: {
+          ...this.getNewVal(value, valType),
+        },
+      }),
+      () => {
+        const { value: updatedVal } = this.state;
+        answerHandler(id, updatedVal, 'change');
+      }
+    );
+  };
+
+  render() {
+    const { question, answer, activeId } = this.props;
+    const {
+      hasFocus,
+      answerable,
+      value: { mass, velocity, kineticEnergy },
+    } = this.state;
+
+    const { id, questionType, label, placeholder } = question;
+    const isTextArea = questionType === 'textArea';
+    const active = activeId === id;
+    const answered = !isEmpty(answer);
+    const cardClasses = classnames('qa-card', {
+      active: hasFocus,
+    });
+    const fieldClasses = classnames('qa-text-input', {
+      answered,
+      unanswered: !answered,
+      answerable: answerable || answered || active,
+    });
+
+    return (
+      <Card
+        className={cardClasses}
+        expanded={hasFocus}
+        onExpanderClick={() => {}}
+      >
+        <CardText>
+          <div className={`col-50 ${styles.marginLeft}`}>
+            <TextField
+              id={`text-${isTextArea ? 'area' : 'input'}-${id}`}
+              className={fieldClasses}
+              type="number"
+              min="0"
+              leftIcon={<MassIcon />}
+              label="Enter the Mass (kg)"
+              lineDirection="center"
+              placeholder={placeholder}
+              defaultValue={answered ? mass : null}
+              onBlur={this.handleBlur}
+              onFocus={this.handleFocus}
+              onChange={value => this.handleChange(value, 'mass')}
+              disabled={!(answerable || answered || active)}
+            />
+          </div>
+          <div className={`col-50 ${styles.marginLeft}`}>
+            <TextField
+              id={`text-${isTextArea ? 'area' : 'input'}-${id}`}
```

Comment:
this can just be id={`text-input-${id}`} because you are dictating what style of input is used and it's never gonna be a textArea.  Actually it's never gonna be a text input it's a number input so you could call it that instead if you want

Reply:
updated to `number-input-${id}` per your suggestion.